### ISSUE TITLE
Build fails on NetBSD

### ISSUE THREAD

**Issue by lhagan** [sass/node-sass#1334]
I'm trying to install node-sass on NetBSD 7.0 w/ Node 5.2.0, but have run into a number of errors. Full platform details in [a gist](https://gist.github.com/lhagan/afbde5abf183a1db1473) under the heading [0].

Here's a list of what I've tried and the end result for each. You can find the complete terminal output for each step under the matching numbered heading ([1], [2], etc.) in the [gist](https://gist.github.com/lhagan/afbde5abf183a1db1473).

1. Installing with `npm install node-sass`, the build fails with `return binding.lstat(pathModule._makeLong(path));`. Similar issue reported in #1138.

2. I cloned the `node-sass` repo and get the same error as above when running `npm install`.

3. As a workaround, I tried `/usr/pkg/bin/node scripts/install.js` to eliminate the path issue. That got me to a 404 error as there is no pre-built binding for NetBSD.

4. I tried building from source: `/usr/pkg/bin/node scripts/build.js`. This got pretty far into the build before failing at the linking stage. It's a similar error to #11, but I'm not using `waf` as far as I know.

```
ld: Release/obj.target/binding/src/binding.o: relocation R_X86_64_32S against `_ZTV14CallbackBridgeIP10Sass_ValuePvE' can not be used when making a shared object; recompile with -fPIC
Release/obj.target/binding/src/binding.o: could not read symbols: Bad value
binding.target.mk:114: recipe for target 'Release/obj.target/binding.node' failed
```

This pretty much exhausts my troubleshooting ideas as I'm not sure how (or what) to recompile with -fPIC. Any suggestions on what to try next, or am I out of luck for NetBSD?

**Comment by saper** [comment]
For 1) what does `process.execPath` say from the node command line?

For 3) which compiler are you using? Which C++ libraries were used to build node? What does ldd /path/to/node say?

**Comment by saper** [comment]
I have posted a full build log from my FreeBSD box https://gist.github.com/9986820a7b186de0eb39
( I am using `--nodedir=/usr/local` because FreeBSD port conveniently installs node include files in `/usr/local/include/node`, so `node-gyp` does not need to download it).

And all the symbols are position-independent (due to `-fPIC`): https://gist.github.com/df761a411a85f41eead8

By the way here are my private [FreeBSD ports](https://github.com/saper/ports-exp) - those are used to build the binaries for FreeBSD in an automated way. Please note I am building libsass first as the library and then link it to node-sass statically (to let people just download one binding file).

**Comment by saper** [comment]
Can you check if https://github.com/nodejs/node-gyp/pull/856 fixes it? You might install the `node-gyp` fork before or patch the `addon.gypi` file manually.

**Comment by lhagan** [comment]
Thanks! nodejs/node-gyp#856 did fix the compile issue and I now have node-sass running on my NetBSD image.

I still had the problem described in 1), but this seems like it's a node issue rather than a node-sass issue:

```
$ node -p process.execPath
node
```

**Comment by xzyfer** [comment]
Sounds like there is isn't anything we can do. Can we close this @saper?

**Comment by saper** [comment]
@lhagan we have fixed the `process.execPath` issue with https://github.com/sass/node-sass/pull/1325 Feel free to re-open and comment on  https://github.com/sass/node-sass/issues/1323 if that fix still does not work for you.

**Comment by saper** [comment]
@xzyfer I think this can be considered fully solved once we bump node-gyp dependency to not yet released 3.2.2

**Comment by xzyfer** [comment]
@saper node-gyp@3.3.0 has been released. Patch to bump the min node-gyp in #1386.
